intel-sgx/fortanix-sgx-tools/src/bin/ftxsgx-runner.rs

```diff
@@ -47,6 +48,30 @@ fn catch_sigbus() {
     }
 }
 
+#[cfg(unix)]
+fn catch_sigusr1() {
+    unsafe {
+        extern "C" fn handle_sigusr1(_signo: c_int, _info: *mut siginfo_t, _context: *mut c_void) {
+            eprintln!("SIGUSR1 triggered, thread_id: {:?}", std::thread::current().id());
+            let instruction_ptr = unsafe { (*(_context as *mut ucontext_t)).uc_mcontext.gregs[REG_RIP as usize] as *const u8};
+            const ENCLU: [u8; 3] = [0x0f, 0x01, 0xd7];
+            let is_enclu = ENCLU.iter().enumerate().all(|(idx, v)| {
+                unsafe { *instruction_ptr.offset(idx as isize) == *v }
+            });
+            if is_enclu {
+                // At enclu instruction - force IP to the next instruction after enclu
+                unsafe { (*(_context as *mut ucontext_t)).uc_mcontext.gregs[REG_RIP as usize] += 3 }
+                eprintln!("Enclave thread {:?} hanged and aboarted by the signal", std::thread::current().id());
```

Comment:
hung up and aborted?

Reply:
This is confusing. The thread may not have hanged. In most cases it was just still executing in the enclave.